spec/system/school/customization_spec.rb

```diff
@@ -229,38 +234,58 @@ def image_path(filename)
   end
 
   context 'there are already existing links in the school' do
-    let!(:school_header_link_1) { create :school_link, kind: SchoolLink::KIND_HEADER, school: school, sort_index: 0 }
-    let!(:school_header_link_2) { create :school_link, kind: SchoolLink::KIND_HEADER, school: school, sort_index: 1 }
-    let!(:school_header_link_3) { create :school_link, kind: SchoolLink::KIND_HEADER, school: school, sort_index: 2 }
+    let!(:school_header_link_1) do
+      create :school_link,
+             kind: SchoolLink::KIND_HEADER,
+             school: school,
+             sort_index: 0
+    end
+    let!(:school_header_link_2) do
+      create :school_link,
+             kind: SchoolLink::KIND_HEADER,
+             school: school,
+             sort_index: 1
+    end
+    let!(:school_header_link_3) do
+      create :school_link,
+             kind: SchoolLink::KIND_HEADER,
+             school: school,
+             sort_index: 2
+    end
 
     scenario 'admin updates and changes order of links' do
       sign_in_user school_admin.user, referrer: customize_school_path
       find('button[title="Edit header links"]').click
-      # rearrange links
-      # move first link down
+
       within("div[data-school-link-id='#{school_header_link_1.id}']") do
         expect(page).to have_selector("button[title='Move Up']")
         expect(page).to have_selector("button[title='Move Down']")
 
+        # move first link down
         find("button[title='Move Down']").click
         sleep 0.1
-        expect(SchoolLink.find_by(id: school_header_link_1.id).sort_index).to eq(1)
+        expect(school_header_link_1.reload.sort_index).to eq(1)
```

Comment:
@developedBySJ Note the changes here. You can rely on `reload` method to reload the data in the instance, Also to make the tests more effective, verify data changes in other records that will be affected/not-affected by this operation.